lib/SPIRV/SPIRVUtil.cpp

```diff
@@ -804,8 +804,11 @@ bool getParameterTypes(Function *F, SmallVectorImpl<Type *> &ArgTys,
       HasSret = true;
       unsigned AS = Arg.getType()->getPointerAddressSpace();
       if (auto *STy = dyn_cast<StructType>(Ty))
-        ArgTys.push_back(
-            TypedPointerType::get(GetStructType(STy->getName()), AS));
+        if (STy->hasName())
+          ArgTys.push_back(
+              TypedPointerType::get(GetStructType(STy->getName()), AS));
+        else
+          ArgTys.push_back(TypedPointerType::get(STy, AS));
```

Comment:
I added an:

>  assert(STy==GetStructType(STy->getName())); 

at line 808 and all the LIT tests pass.  Can we remove the if-else and always do line 811?

I also tried simplifying lines 799-816 and just always doing:

> ArgTys.push_back(Arg.getType());

and this also passed LIT testing.

At a minimum we should fuse the cases on lines 800 and 815 since they have identical bodies.

Reply:
> I added an:
> 
> > assert(STy==GetStructType(STy->getName()));
> 
> at line 808 and all the LIT tests pass. Can we remove the if-else and always do line 811?
> 
> I also tried simplifying lines 799-816 and just always doing:
> 
> > ArgTys.push_back(Arg.getType());
> 
> and this also passed LIT testing.

I think the simplifications are too risky.  The code was written intentionally to handle these variations.